Automattic/wp-calypso
# (5P) FSE: Add contextual tips

#### Changes proposed in this Pull Request

This PR adds contextual tips to the Block Inserter Menu. We've defined five tips on this iteration.

<img width="600" alt="Screen Shot 2020-05-11 at 9 29 53 AM" src="https://user-images.githubusercontent.com/77539/81561736-08b33480-936a-11ea-8668-95a1ee57f850.png">

* Add five tips: `CSS`, `themes`, `plugin`, `color` and `header`
* It supports keywords. It means that it's possible to define some words tied to the tip. For instance, the `CSS` tip will appear when exists the `css` or `style` words into the search term.
* i18n: it's possible to translate the tip keywords in order to guarantee i18n. The script will remove duplicated words.
* Gutenberg cross-version support. It checks the experimental Slot as well as the new createInterpolateElement element component in order to guarantee Atomic compatibility.
* User permission: It will check if the user has the `create` - `settings` permission in order to show or not the tip.

#### Testing instructions

* Apply these patch to your testing site. It's FSE, so you will need to run `> yarn build` if you test locally.
* Start to type one of these words in the block searcher. You should see the tip only if the has the admin permissions.
* Try to test with a non-englihs language. In the following, I've tested in Spanish typing `cabecera` which is the translation of `header`.

<img width="262" alt="Screen Shot 2020-05-11 at 9 58 49 AM" src="https://user-images.githubusercontent.com/77539/81564436-1cf93080-936e-11ea-8170-acb21ff73e61.png">


Related issues: https://github.com/WordPress/gutenberg/issues/20196
